graspy/simulations/simulations_corr.py

```diff
@@ -0,0 +1,317 @@
+        are sampled in the diagonal.
+    
+    Returns
+    -------
+    G1: ndarray (n_vertices, n_vertices)
+        Adjacency matrix the same size as P representing a random graph.
+
+    G2: ndarray (n_vertices, n_vertices)
+        Adjacency matrix the same size as P representing a random graph.
+
+    Examples
+    --------
+    >>> np.random.seed(3)
+    >>> n = [3, 3]
+    >>> p = [[0.5, 0.1], [0.1, 0.5]]
+    >>> r = 0.3
+
+    To sample a correlated SBM graph pair based on n, p and R matrices:
+
+    >>> sbm_corr(n, p, r, directed=False, loops=False)
+    (array([[0., 1., 0., 0., 0., 0.],
+        [1., 0., 0., 0., 0., 0.],
+        [0., 0., 0., 0., 0., 0.],
+        [0., 0., 0., 0., 0., 1.],
+        [0., 0., 0., 0., 0., 0.],
+        [0., 0., 0., 1., 0., 0.]]), array([[0., 1., 0., 0., 0., 0.],
+        [1., 0., 0., 1., 1., 0.],
+        [0., 0., 0., 0., 0., 0.],
+        [0., 1., 0., 0., 0., 1.],
+        [0., 1., 0., 0., 0., 0.],
+        [0., 0., 0., 1., 0., 0.]]))
+    """
+    # test input
+    # Check n
+    if not isinstance(n, (list, np.ndarray)):
+        msg = "n must be a list or np.array, not {}.".format(type(n))
+        raise TypeError(msg)
+    else:
+        n = np.array(n)
+        if not np.issubdtype(n.dtype, np.integer):
+            msg = "There are non-integer elements in n"
+            raise ValueError(msg)
+
+    # Check p
+    if not isinstance(p, (list, np.ndarray)):
+        msg = "p must be a list or np.array, not {}.".format(type(p))
+        raise TypeError(msg)
+    else:
+        p = np.array(p)
+        if not np.issubdtype(p.dtype, np.number):
+            msg = "There are non-numeric elements in p"
+            raise ValueError(msg)
+        elif p.shape != (n.size, n.size):
+            msg = "p is must have shape len(n) x len(n), not {}".format(p.shape)
+            raise ValueError(msg)
+        elif np.any(p < 0) or np.any(p > 1):
+            msg = "Values in p must be in between 0 and 1."
+            raise ValueError(msg)
+
+    # check r
+    if not np.issubdtype(type(r), np.floating):
+        raise TypeError("r is not of type float.")
+    elif r < -1 or r > 1:
+        msg = "r must between -1 and 1."
+        raise ValueError(msg)
+
+    # limit the relation between r and p
+    for i in range(np.array(p).shape[0]):
+        for j in range(np.array(p).shape[1]):
+            if p[i][j] + r * (1 - p[i][j]) < 0:
+                msg = "p + r * (1 - p) should be bigger than 0"
+                raise ValueError(msg)
+
+            elif p[i][j] * (1 - r) < 0:
+                msg = "p * (1 - r) should be bigger than 0"
+                raise ValueError(msg)
+
+    # check directed and loops
+    if type(directed) is not bool:
+        raise TypeError("directed is not of type bool.")
+    if type(loops) is not bool:
+        raise TypeError("loops is not of type bool.")
+
+    P = np.zeros((np.sum(n), np.sum(n)))
+    block_indices = np.insert(np.cumsum(np.array(n)), 0, 0)
+    p = np.mat(p)
```

Comment:
why is this necessary?

Reply:
This is for the next few lines of making a generated P matrix to have the dimension of (n[0] + n[1]) * (n[0] + n[1]).

Reply:
I don't think you need to do `p = np.mat(p)` ?

Reply:
Sure. I can fix it.